docs/DeveloperGuide.md

```diff
@@ -588,31 +590,169 @@ testers are expected to do more *exploratory* testing.
+         - Testing for: Erroneous index results in command failure
+         - Expected: Similar to test case 2.
+
+### Editing data of a contact
+1. Edit data of contact.
+    1. Prerequisites: Non-empty address-book. Navigate to the `Address book` tab using `tab 1`
+    2. Test Cases
+        1. Test case 1: `edit N n/Pierce`, `edit N a/Serangoon Central`, ... where N is a number between 1 and the number of contacts in the list
+            - Testing for: Successful command execution
+            - Expected: Field of first contact updates to that specified in the command
+        2. Test case 2: `edit 1 n/@#`, `edit 1 e/123`, ...
+            - Testing for: Invalid fields results in command failure
+            - Expected: Contacts do not update. Error details shown in the status message.
+        3. Test case 3: `edit 1 x n/Pierce`, ... where x is anything that does not have a field specifier and is not a number between 1 and the number of contacts in the list
+            - Testing for: Unspecified field results in command failure
+            - Expected: Contacts do not update. Error details shown in the status message.
+        4. Test case 4: `edit M n/Pierce`, ... where M is a number not between 1 and the number of contacts in the list
+            - Testing for: Unspecified field results in command failure
+            - Expected: Contacts do not update. Error details shown in the status message.
+        5. Test case 4: `edit`
+            - Testing for: No fields specified results in command failure
+            - Expected: User does not update. Error details shown in the status message.
+
+### Finding a person
+1. Finding a person while all persons are being shown
+   1. Prerequisites: List all persons using the `list` command. Multiple persons in the list. Navigate to the `Address book` tab using `tab 1`
+   2. Test Cases
+      1. Test case 1: `find n/Pi`
+         - Testing for: Command successful execution 
+         - Expected: Shows all contacts with `Pi` in their names show up in the contacts list.
+      2. Test case 2: `find n/Stan n/Pierce`
+        <br>
+         - Testing for: Command successful execution
+         - Expected: Shows all contacts with `Stan` **or** `Pierce` in their names will show up in the contacts list
+      3. Test case 3: `find n/Pie m/Comp`
+        <br>
+         - Testing for: Command successful execution
+         - Expected: Shows all contacts with `Pie` in their name field **and** have `Comp` in their major field.
+      4. Test case 4: `find`
+        <br>
+         - Testing for: No fields specified results in command failure
+         - Expected: List does not update. Error details shown in the status message.
+      5. Test case 5: `find 1 n/Pi`
+         <br>
+         - Testing for: Index specified results in command failure
+         - Expected: List does not update. Error details shown in the status message.
```

Comment:
These don't seem to be rendering properly in the rich diff view, might be because of the indentation of the `<br>`.